src/__spec_helper__/test-utils.js

```diff
@@ -98,6 +100,58 @@ const assertHoverTraversal = assertCorrectTraversal(wrapper => hoverList(wrapper
 
 const carbonThemesJestTable = carbonThemeList.map(theme => [theme.name, theme]);
 
+const marginProps = [
+  ['m', 'margin'],
+  ['ml', 'marginLeft'],
+  ['mr', 'marginRight'],
+  ['mt', 'marginTop'],
+  ['mb', 'marginBottom'],
+  ['mx', 'marginLeft'],
+  ['mx', 'marginRight'],
+  ['my', 'marginTop'],
+  ['my', 'marginBottom']
+];
+
+const paddingProps = [
+  ['p', 'padding'],
+  ['pl', 'paddingLeft'],
+  ['pr', 'paddingRight'],
+  ['pt', 'paddingTop'],
+  ['pb', 'paddingBottom'],
+  ['px', 'paddingLeft'],
+  ['px', 'paddingRight'],
+  ['py', 'paddingTop'],
+  ['py', 'paddingBottom']
+];
+
+const testStyledSystemSpacing = (component) => {
+  describe.each(marginProps)('when a custom margin is specified using the "%s" styled system props',
+    (styledSystemProp, propName) => {
+      it(`then that ${propName} should have been set correctly`, () => {
+        const props = { [styledSystemProp]: 2 };
```

Comment:
As we use the same value for each test, we should add a negative assertion. It would have to have a way to provide the component default too. Without that we wouldn't know if the styled-system prop was being used, or if it just has a default value of 16px.

Reply:
To reduce the chance of this, maybe we should add `space` to the end of the StyledComponent, instead of on the first line.

Reply:
Assuming a default mb of 8px maybe we could have this syntax

`testStyledSystemSpacing(component, {mb: "8px"})`